aiida_widgets/structures.py

```diff
@@ -59,21 +58,26 @@ def _on_file_upload(self, change):
         with self.output:
             clear_output()
 
-            tmp = tempfile.mkdtemp() + '/' + self.file_upload.filename
+            self.tmp_folder = tempfile.mkdtemp()
+            tmp = self.tmp_folder + '/' + self.file_upload.filename
             with open(tmp, 'w') as f:
                 f.write(self.file_upload.data)
             self.tmp_file = tmp
 
-            traj = ase.io.read(tmp, index=":")
-            if len(traj) > 1:
-                print("Error: Uploaded file contained more than one structure")
-            atoms = traj[0]
-            formula = atoms.get_chemical_formula()
-            self.structure_label.value = "{} ({})".format(
-                formula, self.file_upload.filename)
-            self.atoms = atoms
-            self.refresh_view()
-            self.btn_store.disabled = False
+            self.select_structure(name=self.file_upload.filename)
+
+    def select_structure(self, name=None):
+        self.tmp_file = self.tmp_folder + '/' + name
```

Comment:
function has parameter name=None, but this won't work with name=None

Reply:
Ok, I deleted the default value